go/obscuronode/enclave/enclave.go

```diff
@@ -120,54 +123,58 @@ func (e *enclaveImpl) ProduceGenesis(blkHash common.Hash) nodecommon.BlockSubmis
 	}
 }
 
+// IngestBlocks is used to update the enclave with the full history of the L1 chain to date.
 func (e *enclaveImpl) IngestBlocks(blocks []*types.Block) []nodecommon.BlockSubmissionResponse {
 	result := make([]nodecommon.BlockSubmissionResponse, len(blocks))
 	for i, block := range blocks {
+		// We skip over the genesis block, to avoid an attack whereby someone submits a block with the same hash as the
+		// genesis block but different contents. Since we cannot insert the genesis block into our blockchain, this
+		// checking would have to be skipped, potentially allowing an invalid block through.
+		if e.isGenesisBlock(block) {
+			continue
+		}
+
+		if ingestionFailedResponse := e.insertBlockIntoL1Chain(block); ingestionFailedResponse != nil {
+			result[i] = *ingestionFailedResponse
+			continue
```

Comment:
This should be a return, I think. If a block has failed, the descendants will fail for sure.

Reply:
+1